shared/src/androidMain/kotlin/org/hyperskill/app/purchases/domain/AndroidPurchaseManager.kt

```diff
@@ -0,0 +1,110 @@
+package org.hyperskill.app.purchases.domain
+
+import android.app.Activity
+import android.app.Application
+import com.revenuecat.purchases.LogLevel
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.PurchasesConfiguration
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.PurchasesException
+import com.revenuecat.purchases.PurchasesTransactionException
+import com.revenuecat.purchases.awaitCustomerInfo
+import com.revenuecat.purchases.awaitGetProducts
+import com.revenuecat.purchases.awaitLogIn
+import com.revenuecat.purchases.awaitLogOut
+import com.revenuecat.purchases.awaitPurchase
+import com.revenuecat.purchases.models.StoreProduct
+import org.hyperskill.app.BuildConfig
+import org.hyperskill.app.purchases.domain.model.AndroidPurchaseParams
+import org.hyperskill.app.purchases.domain.model.PlatformPurchaseParams
+import org.hyperskill.app.purchases.domain.model.PurchaseManager
+import org.hyperskill.app.purchases.domain.model.PurchaseResult
+
+class AndroidPurchaseManager(
+    private val application: Application,
+    private val isDebugMode: Boolean
+) : PurchaseManager {
+    override fun setup() {
+        if (!Purchases.isConfigured) {
+            Purchases.logLevel = if (isDebugMode) LogLevel.DEBUG else LogLevel.INFO
+            Purchases.configure(
+                PurchasesConfiguration.Builder(
+                    context = application,
+                    apiKey = BuildConfig.REVENUE_CAT_GOOGLE_API_KEY
+                ).build()
+            )
+        }
+    }
+
+    override suspend fun login(userId: Long): Result<Unit> =
+        kotlin.runCatching {
+            Purchases.sharedInstance.awaitLogIn(userId.toString())
+        }
+
+    override suspend fun logout(): Result<Unit> =
+        runCatching {
+            Purchases.sharedInstance.awaitLogOut()
+        }
+
+    override suspend fun purchase(
+        productId: String,
+        platformPurchaseParams: PlatformPurchaseParams
+    ): Result<PurchaseResult> =
```

Comment:
`Result<PurchaseResult>` as for me this is strange design that we returns doubled return value - first result can be successful or not and the second result the same. Ideally, I would expect just `PurchaseResult`.